bundles/mapping/mapmodule/plugin/markers/MarkersPlugin.ol.js

```diff
@@ -148,315 +117,182 @@ Oskari.clazz.define('Oskari.mapframework.mapmodule.MarkersPlugin',
-         * @param e
-         * @private
+         * Activate the "add marker mode" on map.
          */
-        _showForm: function (clickX, clickY) {
-            var me = this;
-            // if we dont set false here the user can click map again and a new popup is opened on top of the existing one
-            me._waitingUserClickToAddMarker = false;
-            var lonlat = me._map.getCoordinateFromPixel([
-                clickX,
-                clickY
-            ]);
-            var loc = me.getLocalization().form;
-            me.dotForm = Oskari.clazz.create(
-                'Oskari.userinterface.component.visualization-form.DotForm',
-                me,
-                loc,
-                me._defaultData
-            );
-            var dialog = me.dotForm.getDialog();
-            if (dialog) {
-                dialog.close(true);
-            }
-            me.dotForm.showForm(jQuery('div.selection-line')[0], {
-                messageEnabled: true
-            }, 'right');
-
-            me.dotForm.setSaveHandler(function () {
-                var values = me.dotForm.getValues();
-                var reqBuilder = Oskari.requestBuilder(
-                    'MapModulePlugin.AddMarkerRequest'
-                );
-
-                if (reqBuilder) {
-                    var data = {
-                        x: lonlat[0],
-                        y: lonlat[1],
-                        msg: values.message,
-                        color: values.color,
-                        shape: values.shape,
-                        size: values.size
-                    };
-                    var request = reqBuilder(data);
-                    me.getSandbox().request(me.getName(), request);
-                    me.stopMarkerAdd(true);
-                }
-                me.dotForm.getDialog().close(true);
-                me.enableGfi(true);
-            });
+        startMarkerAdd: function () {
+            this.enableGfi(false);
+            this._showAddMarkerPopupOnMapClick = true;
+            const clearBtn = Oskari.clazz.create('Oskari.userinterface.component.Button');
+            const closeBtn = Oskari.clazz.create('Oskari.userinterface.component.buttons.CloseButton');
+
+            clearBtn.setTitle(this.getMsg('dialog.clear'));
+            clearBtn.setHandler(() => this.removeMarkers());
+            closeBtn.setHandler(() => this.stopMarkerAdd(true));
+            closeBtn.setPrimary(true);
 
-            me.dotForm.setCancelHandler(function () {
-                // return to wait another click for a marker
-                me.dotForm.getDialog().close();
-                me._waitingUserClickToAddMarker = true;
-            });
+            this.dialog.show(
+                this.getMsg('title'),
+                this.getMsg('dialog.message'),
+                [clearBtn, closeBtn]
+            );
+            this.getMapModule().getMapEl().addClass(
+                'cursor-crosshair'
+            );
```

Comment:
Should we use `this.getMapModule().setCursorStyle('crosshair')` instead since that's available? Not sure where `.cursor-crosshair` is defined in css but if it's on the marker plugin css we could remove that as well.